### ISSUE TITLE
How can I get block object based on ID?

### ISSUE THREAD

**Issue by jonathanlundstrom** [AdvancedCustomFields/acf#639]
Hi,

I'm looping through some fields that are part of ACF Gutenberg blocks and I want to fetch the block object so that I can access information such as icon, title, description etc. If I know the block ID, such as `block_612de94a30387 `, how can I go about fetching the object? I tried to use `acf_get_block_types()` but none of the items in the array have an ID that I can compare against.

Any help is appreciated!

Regards,
Jonathan

**Comment by lgladdy** [comment]
Hey Jonathan,

A block ID has no correlation to the type of block, instead you'd need to use `$block['name']` which contains the block's name including the automatic `acf/` namespace.

**Comment by jonathanlundstrom** [comment]
Hi @lgladdy, thanks for getting back to me. I was a bit tired at the point of writing this, and looking back I understand that the ID cannot be traced back to the block. I've attached an image below to display what I'm trying to accomplish.

![image](https://user-images.githubusercontent.com/4216027/163381287-051fd5f9-1ad1-4ea9-b6f6-686ef7f4f1c3.png)

Currently I'm using the filter `acf/pre_render_fields` to get the output where I want it, but at this point I have no access to the actual `$block` array, so I cannot fetch the information I'm looking for (icon, title and description). I've been going through all the source files and documentation but I've not been able to find a better hook in order to display content in the actual edit mode of the block.

I also tried to access the first field, get its group location and then use the block name to fetch the `$block` variable. The only problem is that some fields are shared between different groups (included through PHP files) and this means the parent isn't necessarily the parent group of the current block, but can rather point to another block, at which point the block I'm fetching is not of the correct type.

Do you have any ideas of other hooks, or how I can access the block object from within the hook?

**Comment by lgladdy** [comment]
Hey Jonathan,

I can't think of a way this is possible at the moment. That edit form is only ever loaded by AJAX, and it's not designed to have any awareness of the block that is currently being displayed in at the point the field is rendered.

Looking up the chain, the nearest filter is: `acf/load_field_groups` which is called when we try to load the field groups and that will have an array passed to it with the block name identifying we're trying to load the block's field groups... You could potentially when inject a new prepending field group, which contained a message field, which contained all the html content you wanted to display... but obviously that's not the cleanest solution to the problem.

**Comment by jonathanlundstrom** [comment]
Yeah, I've been down that road as well, but as you say, injecting an extra field doesn't seem like a clean solution. To be fair, I really only want to create a simple PoC to show you guys. I've got an idea on how to improve the overall experience of editing pages with tons of ACF Blocks. I was hoping I could do it without a PR since I'm low on time at the moment, but I guess I could always create a PR for the changes I think should be implemented in core 😄 

Thanks, and if you do think of something, please let me know! 👍

**Comment by foldesistudio** [comment]
My suggestion is to create a helper function for this task. Here's my proposed solution:
`
/**
 * Get ACF Block name for class.
 *
 * @param array $block The block array containing 'name' key.
 * @return string The second part of the split string, or an empty string if the block name does not contain '/'.
 */
function getBlockName($block) {
	$blockName = $block['name'];
	$parts = explode('/', $blockName);
	return $parts[1] ?? '';
}

`
